File: src/lib/types/audit.ts
/**
 * This module defines typed front-end contracts for audit-ledger review, doctor findings, and repair reports.
 *
 * Why this file exists:
 * - The UI reads these shapes as its desktop and preview contract, so unclear names here ripple through every consumer.
 * - If you need to know what a route or helper expects from the backend, this is often the fastest file to open first.
 *
 * Main declarations:
 * - `AuditArtifact`
 * - `AuditRunDetail`
 * - `HealthCheck`
 * - `HealthReport`
 * - `HealthRepairReport`
 *
 * Source-of-truth notes:
 * - Data shapes should stay aligned with the accepted architecture and feature docs rather than ad-hoc page assumptions.
 * - Prefer additive, explicit fields over ambiguous catch-all objects so the trust surface stays auditable.
 */

import type { BackupRunOverview } from './archive'

/**
 * Defines the typed shape for audit artifact.
 *
 * These type contracts are read directly by routes, helper modules, and preview fixtures, so a reader should be able to understand the shape without hunting through call sites.
 */
export interface AuditArtifact {
  kind: string
  path: string
  checksum?: string | null
  sizeBytes?: number | null
  createdAt: string
  reason?: string | null
}

/**
 * Represents the detailed view model for audit run.
 *
 * These type contracts are read directly by routes, helper modules, and preview fixtures, so a reader should be able to understand the shape without hunting through call sites.
 */
export interface AuditRunDetail {
  run: BackupRunOverview
  trigger: string
  timezone?: string | null
  dueOnly: boolean
  profileScope: string[]
  warnings: string[]
  errorMessage?: string | null
  stats: Record<string, unknown>
  manifestPath?: string | null
  manifestHash?: string | null
  artifacts: AuditArtifact[]
}
/**
 * Defines the typed shape for health check.
 *
 * These type contracts are read directly by routes, helper modules, and preview fixtures, so a reader should be able to understand the shape without hunting through call sites.
 */
export interface HealthCheck {
  name: string
  status: string
  message: string
}

/**
 * Represents a completed report that the UI can review after a run finishes.
 *
 * These type contracts are read directly by routes, helper modules, and preview fixtures, so a reader should be able to understand the shape without hunting through call sites.
 */
export interface HealthReport {
  generatedAt: string
  checks: HealthCheck[]
}

/**
 * Represents a completed report that the UI can review after a run finishes.
 *
 * These type contracts are read directly by routes, helper modules, and preview fixtures, so a reader should be able to understand the shape without hunting through call sites.
 */
export interface HealthRepairReport {
  runId?: number | null
  repairedImportAudits: number
  repairedVisibilityRows: number
  clearedDerivedRows: number
  notes: string[]
}

File: src-tauri/crates/vault-core/src/models/audit.rs
use serde::{Deserialize, Serialize};
use serde_json::Value;
use super::BackupRunOverview;

#[derive(Debug, Clone, Serialize, Deserialize, Default)]
#[serde(rename_all = "camelCase")]
pub struct AuditArtifact {
    pub kind: String,
    pub path: String,
    pub checksum: Option<String>,
    pub size_bytes: Option<u64>,
    pub created_at: String,
    pub reason: Option<String>,
}

#[derive(Debug, Clone, Serialize, Deserialize, Default)]
#[serde(rename_all = "camelCase")]
pub struct AuditRunDetail {
    pub run: BackupRunOverview,
    pub trigger: String,
    pub timezone: Option<String>,
    pub due_only: bool,
    pub profile_scope: Vec<String>,
    pub warnings: Vec<String>,
    pub error_message: Option<String>,
    pub stats: Value,
    pub manifest_path: Option<String>,
    pub manifest_hash: Option<String>,
    pub artifacts: Vec<AuditArtifact>,
}
#[derive(Debug, Clone, Serialize, Deserialize)]
#[serde(rename_all = "camelCase")]
pub struct HealthCheck {
    pub name: String,
    pub ok: bool,
    pub detail: String,
}

#[derive(Debug, Clone, Serialize, Deserialize, Default)]
#[serde(rename_all = "camelCase")]
pub struct HealthReport {
    pub generated_at: String,
    pub checks: Vec<HealthCheck>,
}

#[derive(Debug, Clone, Serialize, Deserialize, Default)]
#[serde(rename_all = "camelCase")]
pub struct HealthRepairReport {
    pub run_id: Option<i64>,
    pub repaired_import_audits: usize,
    pub repaired_visibility_rows: usize,
    pub cleared_derived_rows: usize,
    pub notes: Vec<String>,
}
